src/components/grid.js

```diff
@@ -86,87 +122,109 @@ class Grid extends Component {
 
-        if (inventory === 0) {
+        if (!inventory) {
             this.props.updateMessage(
-                'You must first bring me back 3 report cards before you are allowed to pass!'
+                `"Greetings, ${name}! You must first bring me back 3 report cards before you are allowed to pass!"`,
+                'dialogue'
             );
             return false;
-        } else if (inventory < required) {
+        } else if (inventory < amountRequired) {
             this.props.updateMessage(
-                `It looks like you're almost there! Only ${
-                    required - inventory[0].quantity
-                } more report card${
-                    inventory[0].quantity !== 1 ? '' : 's'
-                } to go!`
+                `"Oh, you're quite on your way, ${name}! You only need ${
+                    amountRequired - inventory
+                } more report card${inventory !== 1 ? '' : 's'}!"`,
+                'dialogue'
             );
             return false;
         }
 
         return true;
     }
 
-    interact(row, column) {
+    handleInteraction(row, column) {
         const gridSpace = this.state.gridLayout[row][column];
+        const { updateMessage } = this.props;
 
         if (gridSpace.npc && !gridSpace.npc.isInTheMiddleOfAction) {
             const currentAction = gridSpace.npc.interact();
+            const cancel = gridSpace.npc.cancel();
 
             if (currentAction.requirements.length) {
                 for (let require of currentAction.requirements) {
-                    const {
-                        type,
-                        item,
-                        numberNeeded,
-                        question,
-                        answer,
-                        cleared,
-                    } = require;
+                    const { type, item, amount, question, answer, cleared } =
+                        require;
 
                     if (type === RequirementTypes.inventory) {
-                        if (!this.checkPlayerInventory(item, numberNeeded))
+                        if (!this.checkPlayerInventory(item, amount)) {
+                            cancel;
                             return false;
-                        // if (!Object.hasOwnProperty.call(player, type) || player[type] < amount) {
-                        //     return {
-                        //         success: false,
-                        //         message,
-                        //     };
-                        // }
+                        }
                     } else if (type === RequirementTypes.question) {
                         const playerAnswer = this.props.answerInput;
-                        this.props.clearAnswerInput();
+
                         if (cleared) {
-                            console.log('already answered');
-                            return;
+                            cancel;
```

Comment:
What's the purpose of all these `cancel` statements?

Reply:
You're right, it's not needed here. Also fixed the below comment regarding the `cancel` variable storing a reference to the cancel method.